Sources/Starknet/Providers/StarknetProviderProtocol.swift

```diff
@@ -2,275 +2,31 @@ import Foundation
-    ///
-    ///  - Returns: array of simulated transactions
-    func simulateTransactions(_ transactions: [any StarknetExecutableTransaction], at blockId: StarknetBlockId, simulationFlags: Set<StarknetSimulationFlag>) -> StarknetRequest<[StarknetSimulatedTransaction]>
-
-    /// Batch multiple calls into a single RPC request
+    /// Send RPC request
     ///
     /// - Parameters
-    ///     - requests: list of requests to be batched together.
+    ///     - request: single request to be sent.
     ///
-    /// - Returns: batch request.
-    func batchRequests<U: Decodable>(requests: [StarknetRequest<U>]) throws -> StarknetBatchRequest<U>
+    /// - Returns: request result.
+    func send<U>(request: StarknetRequest<U>) async throws -> U
 
-    /// Batch multiple calls into a single RPC request
+    /// Send multiple calls in a single RPC request
     ///
     /// - Parameters
-    ///     - requests: requests to be batched together.
-    ///
-    /// - Returns: batch request.
-    func batchRequests<U: Decodable>(requests: StarknetRequest<U>...) throws -> StarknetBatchRequest<U>
-}
-
-let defaultBlockId = StarknetBlockId.tag(.pending)
-let defaultSimulationFlagsForEstimateFee: Set<StarknetSimulationFlagForEstimateFee> = []
-public extension StarknetProviderProtocol {
-    /// Call starknet contract in the pending block.
-    ///
-    /// - Parameters
-    ///     - call: starknet call to be made.
-    ///
-    /// - Returns: Array of field elements, returned by called contract.
-    func callContract(_ call: StarknetCall) -> StarknetRequest<[Felt]> {
-        callContract(call, at: defaultBlockId)
-    }
-
-    /// Estimate fee for a list of transactions with default flags in the pending block.
-    ///
-    /// - Parameters:
-    ///  -  transactions: transactions for which the fees should be estimated.
-    ///
-    /// - Returns: Array of fee estimates
-    func estimateFee(for transactions: [any StarknetExecutableTransaction]) -> StarknetRequest<[StarknetFeeEstimate]> {
-        estimateFee(for: transactions, at: defaultBlockId, simulationFlags: defaultSimulationFlagsForEstimateFee)
-    }
-
-    /// Estimate fee for a list of transactions with default flags.
-    ///
-    /// - Parameters:
-    ///  -  transactions: transactions for which the fees should be estimated.
-    ///  -  blockId: hash, numer, or tag of a block for which the estimation should be made.
+    ///     - requests: list of requests to be sent.
     ///
-    /// - Returns: Array of fee estimates
-    func estimateFee(for transactions: [any StarknetExecutableTransaction], at blockId: StarknetBlockId) -> StarknetRequest<[StarknetFeeEstimate]> {
-        estimateFee(for: transactions, at: blockId, simulationFlags: defaultSimulationFlagsForEstimateFee)
-    }
+    /// - Returns: requests' results.
+    func send<U>(
+        requests: [StarknetRequest<U>]
+    ) async throws -> [Result<U, StarknetProviderError>] where U: Decodable
```

Comment:
IMO it's fine. To me it's clear that if you are sending an array of requests, then it's a batch.

Reply:
> I'm not sure following this change there's any indication from the function name or the logic that there requests are sent in one batch, not for example sequentially.

In function DocC there is `Send multiple calls in a single RPC request` which covers the fact that calls are sent in one request. I think having `send` both for single/batch request is more elegant. Also, if we had two methods, sending batch request and requests sequence respectively, then this would really make sense.

If we want to change it, we could rename it to e.g. `sendBatch`. 

Wdyt about both solutions @DelevoXDG ?